synapse/storage/databases/main/media_repository.py

```diff
@@ -126,13 +149,42 @@ async def get_local_media_by_user_paginate(
             start: offset in the list
             limit: maximum amount of media_ids to retrieve
             user_id: fully-qualified user id
+            order_by: the sort order of the returned list
+            direction: sort ascending or descending
         Returns:
             A paginated list of all metadata of user's media,
             plus the total count of all the user's media
         """
 
         def get_local_media_by_user_paginate_txn(txn):
 
+            # Set ordering
+            if MediaSortOrder(order_by) == MediaSortOrder.MEDIA_ID:
+                order_by_column = "media_id"
+            elif MediaSortOrder(order_by) == MediaSortOrder.UPLOAD_NAME:
+                order_by_column = "upload_name"
+            elif MediaSortOrder(order_by) == MediaSortOrder.CREATED_TS:
+                order_by_column = "created_ts"
+            elif MediaSortOrder(order_by) == MediaSortOrder.LAST_ACCESS_TS:
+                order_by_column = "last_access_ts"
+            elif MediaSortOrder(order_by) == MediaSortOrder.MEDIA_LENGTH:
+                order_by_column = "media_length"
+            elif MediaSortOrder(order_by) == MediaSortOrder.MEDIA_TYPE:
+                order_by_column = "media_type"
+            elif MediaSortOrder(order_by) == MediaSortOrder.QUARANTINED_BY:
+                order_by_column = "quarantined_by"
+            elif MediaSortOrder(order_by) == MediaSortOrder.SAFE_FROM_QUARANTINE:
+                order_by_column = "safe_from_quarantine"
+            else:
+                raise StoreError(
+                    500, "Incorrect value for order_by provided: %s" % order_by
+                )
```

Comment:
This isn't really a storage-specific error: I think ValueError is more appropriate.

```suggestion
                raise ValueError(
                    "Incorrect value for order_by provided: %s" % (order_by,)
                )
```